### PR TITLE
Cog1 qm bypass fix and new remote door control button functionality

### LINKED ISSUE #1307: Cog 1 QM Bypass belts configured wrong
Submitter: coreanus
**Describe the bug**
In Cog1 QM there is a button to route input from belt hell straight back (Button label: safety shutters)
This closes 2 shutters leading into QM, opens one inbetween the belts, and switches the On/Off status of a west-facing belt and a South facing belt on the incoming line respectiveley. However, the initial states of these belt are wrong, leading to the west-facing belt being on when the shutters are open, and the north-facing one being on when they are closed. This makes crates get stuck.

**To Reproduce**
Press button
See error
*scream

**Expected behavior**
The belts actually let crates in/cycle them back as intended

Reported on: Goonstation Roleplay 2020-07-02 12:04:49

### PR DESCRIPTION
**Pull request by eX-n0x**
[BUG] [FEATURE] [MAPPING] [QOL]
## About the PR 
This fixes the Cog1 Cargo Bypass Button by adding functionality that allows remote door control buttons to have cooldowns that stops spamming which breaks whatever they are attached to. This can be used on any other remote door control buttons that mappers want as well by adding whatever number they want on the "cooldown" var which works in seconds, (e.g. 1 = 1 second).

## Why's this needed? 
Fixes #1307 and would be nice to have the extra functionality just in case.

### CODE REVIEW

**Comment by ZeWaka** [comment]
No changelig necessary.

File: code/obj/machinery/door/door_control.dm [hunk 1]
```diff
@@ -447,6 +449,11 @@
 						M.operating = 0
 			M.setdir()
 
+	if(src.cooldown)
+		inuse = 1
+		SPAWN_DBG(src.cooldown SECONDS)
+		inuse = 0
```
Comment (ZeWaka) [inline code/obj/machinery/door/door_control.dm:455]: This is not how spawns work

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/obj/machinery/door/door_control.dm [hunk 2]
```diff
@@ -447,6 +449,11 @@
 						M.operating = 0
 			M.setdir()
 
+	if(src.cooldown)
+		inuse = 1
```
Comment (ZeWaka) [inline code/obj/machinery/door/door_control.dm:453]:
```suggestion
		inuse = TRUE
```

File: code/obj/machinery/door/door_control.dm [hunk 3]
```diff
@@ -447,6 +449,11 @@
...
+		sleep(src.cooldown SECONDS)
+		inuse = 0
```
Comment (ZeWaka) [inline code/obj/machinery/door/door_control.dm:455]:
```suggestion
		inuse = FALSE
```

File: code/obj/machinery/door/door_control.dm [hunk 4]
```diff
@@ -5,6 +5,8 @@
 	desc = "A remote control switch for a door."
 	var/id = null
 	var/timer = 0
+	var/cooldown = 0
+	var/inuse = 0
```
Comment (ZeWaka) [inline code/obj/machinery/door/door_control.dm:9]:
```suggestion
	var/inuse = FALSE
```

File: code/obj/machinery/door/door_control.dm [hunk 5]
```diff
@@ -5,6 +5,8 @@
 	desc = "A remote control switch for a door."
 	var/id = null
 	var/timer = 0
+	var/cooldown = 0
```
Comment (ZeWaka) [inline code/obj/machinery/door/door_control.dm:8]:
```suggestion
	var/cooldown = 0 SECONDS
```

**Review by ZeWaka — COMMENTED** [review]
Units grr

File: code/obj/machinery/door/door_control.dm (hunk 5 above)
Comment (eX-n0x) [reply code/obj/machinery/door/door_control.dm]: Speaking of which, would this be redundant to have it as seconds both on the var and the sleep function?

**Review by ZeWaka — APPROVED** [review]

File: code/obj/machinery/door/door_control.dm (hunk 5 above)
Comment (ZeWaka) [reply code/obj/machinery/door/door_control.dm]: yes. very.

File: code/obj/machinery/door/door_control.dm [hunk 6]
```diff
@@ -447,6 +449,11 @@
 						M.operating = 0
 			M.setdir()
 
+	if(src.cooldown)
+		inuse = TRUE
+		sleep(src.cooldown SECONDS)
```
Comment (ZeWaka) [inline code/obj/machinery/door/door_control.dm:454]:
```suggestion
		sleep(src.cooldown)
```

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**